src/pages/api/auth/[...nextauth].ts

```diff
@@ -0,0 +1,40 @@
+import NextAuth from "next-auth";
+import GoogleProvider from "next-auth/providers/google";
+import CredentialsProvider from 'next-auth/providers/credentials';
+
+
+export const authOptions = {
+  // Configure one or more authentication providers
+  providers: [
+    GoogleProvider({
+        clientId: process.env.GOOGLE_CLIENT_ID as string,
+        clientSecret: process.env.GOOGLE_CLIENT_SECRET as string
+      }),
+      CredentialsProvider({
+        name: "Credentials",
+        credentials: {
+            username: {
+                label: "Username:",
+                type: "text",
+                placeholder: "your-cool-username"
+            },
+            password: {
+                label: "Password:",
+                type: "password",
+                placeholder: "your-awesome-password"
+            }
+        },
+        async authorize(credentials) {
+            const user = { id: "42", name: "Dave", password: "nextauth" }
```

Comment:
I going to keep this name of use like this for this time because it will cause a lot of. changes in two branches that l already created a pull request for. Will keep this note in mind for the future thank you for sharing